File: leetcode/0028-find-the-index-of-the-first-occurrence-in-a-string/solution.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        n, m = len(haystack), len(needle)
        for i in range(n - m + 1):
            if haystack[i : i + m] == needle:
                return i
        return -1


if __name__ == "__main__":
    print(Solution().strStr("sadbutsad", "sad"))  # 0
    print(Solution().strStr("leetcode", "leeto"))  # -1

File: leetcode/0028-find-the-index-of-the-first-occurrence-in-a-string/solution.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int strStr(string haystack, string needle) {
        int n = (int)haystack.size(), m = (int)needle.size();
        for (int i = 0; i + m <= n; i++) {
            bool match = true;
            for (int j = 0; j < m; j++) {
                if (haystack[i + j] != needle[j]) {
                    match = false;
                    break;
                }
            }
            if (match) return i;
        }
        return -1;
    }
};

int main() {
    Solution sol;
    cout << sol.strStr("sadbutsad", "sad") << endl;   // 0
    cout << sol.strStr("leetcode", "leeto") << endl;   // -1
    return 0;
}